src/runtime/app.ts

```diff
@@ -36,6 +36,26 @@ function createNitroApp (): NitroApp {
 
   const routerOptions = createMatcher({ routes: config.nitro.routes })
 
+  h3App.use(eventHandler((event) => {
+    const routeOptions = routerOptions.lookup(event.req.url) || {}
+    // Share applicable route rules across handlers
+    event.context.routeOptions = routeOptions
+    if (routeOptions.cors) {
+      setHeaders(event, {
+        'access-control-allow-origin': '*',
+        'access-control-allowed-methods': '*',
+        'access-control-allow-headers': '*',
+        'access-control-max-age': '0'
```

Comment:
We might refactor this to h3 later and support cors options (rather than a boolean). Alternatively this could be compiled into headers for better cross platform support